lib/task/import/csvImportTask.class.php

```diff
@@ -956,16 +962,35 @@ public function execute($arguments = array(), $options = array())
         // if both a URI and path are provided, the former is preferred.
         if ($uri = $self->rowStatusVars['digitalObjectURI'])
         {
+          $do = new QubitDigitalObject;
+          $do->informationObject = $self->object;
+
+          if (isset($self->status['options']['skip-derivatives']))
+          {
+            // If not creating derivatives, then skip download altogether
+            $options = array('skipDerivatives' => true);
+          }
+          else
+          {
+            // Try downloading external object up to three times (2 retries)
+            $options = array('downloadRetries' => 2);
+          }
+
+          // Catch digital object import errors to avoid killing whole import
           try
           {
-            if (!downloadExternalURIWithRetries($uri, $self->object))
-            {
-              $this->log("Failed to download $uri after multiple tries. Continuing task...");
-            }
+            $do->importFromURI($uri, $options);
+            $do->save($conn);
           }
-          catch (Exception $e)
+          catch (sfException $e)
```

Comment:
Subtle change here but this will cause a range of exceptions not subclassing `sfException` not to be caught, e.g. the library you are consuming may change impl. and bubble up a `InvalidArgumentException` which you may also want to capture in this case...